sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnType.scala

```diff
@@ -102,6 +103,16 @@ private[sql] sealed abstract class ColumnType[JvmType] {
   override def toString: String = getClass.getSimpleName.stripSuffix("$")
 }
 
+private[sql] object NULL extends ColumnType[Any] {
+
+  override def dataType: DataType = NullType
+  override def defaultSize: Int = 0
+  override def append(v: Any, buffer: ByteBuffer): Unit = {}
+  override def extract(buffer: ByteBuffer): Any = null
+  override def setField(row: MutableRow, ordinal: Int, value: Any): Unit = row.setNullAt(ordinal)
+  override def getField(row: InternalRow, ordinal: Int): Any = null
+}
+
```

Comment:
Found that:

1. we can create a DataFrame with `NullType` via `sqlContext.createDataFrame(rdd, schema)` with a schema having a `NullType` column
2. before, all-null columns are materialized using `GENERIC` column type

But I'm not sure whether such columns are useful, since we can't create such columns using either our own SQL dialect or HiveQL.